### DISCUSSION TITLE
Maven clean lifecycle in GitHub actions

### DISCUSSION THREAD

**Discussion by amrsamii** [community/community#26498] [Actions]
<p>Hello,</p>
<p>I am using Maven with GitHub actions. I want to know if the <code>clean</code> lifecycle in the Maven command is useful or not. if the build is starting from scratch everytime maybe it is not needed because there is not <code>target</code> folder anyway. but if I am caching dependencies, is <code>clean</code> still unnecessary?</p>
<p>here is my yml file:</p>
<pre><code>on:
push:
branches: [master]
pull\_request:
branches: [master]

jobs:
build:

runs-on: ubuntu-latest

steps:
-uses: actions/checkout@v2
-name: Set up JDK 1.8
uses: actions/setup-java@v1
with:
java-version: 1.8
-uses: actions/cache@v1
with:
path: ~/.m2/repository
key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
restore-keys: |
${{ runner.os }}-maven-
-name: Build with Maven
run: ./mvnw -B clean verify 
</code></pre>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Cache dependencies has no relationship with clean option.  Cache content under .m2/repository path will not impact the ‘target’ folder . </p>
<p>In your yml file, you are using GitHub hosted runner, each job runs in a fresh instance. So clean is not needed. </p>
<p> There is the official document for Maven build using GitHub Actions <a href="https://help.github.com/en/actions/language-and-framework-guides/building-and-testing-java-with-maven#caching-dependencies" rel="nofollow noopener">https://help.github.com/en/actions/language-and-framework-guides/building-and-testing-java-with-maven#caching-dependencies</a></p>
